scylla/src/statement/batch.rs

```diff
@@ -12,15 +12,16 @@ pub use crate::frame::request::batch::BatchType;
 pub struct Batch {
     pub(crate) config: StatementConfig,
 
-    statements: Vec<BatchStatement>,
+    pub statements: Vec<BatchStatement>,
     batch_type: BatchType,
 }
 
 impl Batch {
-    /// Creates a new, empty `Batch` of `batch_type` type.
-    pub fn new(batch_type: BatchType) -> Self {
+    /// Creates a new, empty `Batch` of `batch_type` type with the statements.
+    pub fn new(batch_type: BatchType, statements: Vec<BatchStatement>) -> Self {
```

Comment:
Note that this change breaks compatibility and would require users to rewrite their code upon upgrade. Which is fine on its own, since the change would be rather trivial, but please do the following:
1. Mention this breaking change in the commit message if you decide it should stay
2. Consider creating a constructor with a different name, and leave `new` as is -- then, compatibility is preserved, and you can still create a batch with a vector of statement, just by using a function with a different name